controllers/ssp_controller.go

```diff
@@ -84,6 +86,11 @@ func CreateAndSetupSspReconciler(mgr ctrl.Manager) error {
 		return err
 	}
 
+	infrastructureTopology, err := common.GetInfrastructureTopology(mgr.GetClient())
```

Comment:
The CI failure is related to this line. When you look into the logs of ssp-operator pod on CI, there is this error:
```
2021-12-06T14:53:57.729Z	ERROR	setup	unable to create controller	{"controller": "SSP", "error": "the cache is not started, can not read objects"}
```

Searching the source code for `"the cache is not started, can not read objects"`, I see that this error is returned by the client cache, when informers were not started. Controller-runtime uses cached client, and the cachce internally uses informers to watch resources and update itself. The cache will be started when the manager is started.

This line of code is executed before the manager is started, so the cache is not running. Instead of `mgr.GetClient()` you should be able to use the uncached API reader `mgr.GetAPIReader()`.

Reply:
Oh, thank you for the explanation!